### DISCUSSION TITLE
New windows sometimes jump to another workspace

### DISCUSSION THREAD

**Discussion by hiredman** [i3/i3#4531] [Q&A]
I have a weird behavior I have been experiencing  for years now.

Sometimes (most often with firefox) and when I create a new window, it starts to render (I see the kind of empty container, and in the firefox case I start to see the chrome), and then the window jumps away to another workspace (I have 3 screens, and it always jumps to a workspace that is not visible). I have not observed any kind of pattern for what non-visible workspace it jumps to.

Anyone have something similar happen, or know of a fix?

**Comment by ghost** [comment]
yes, for me Sublime-Text, Sublime-Merge, and Chrome/ium have special needs where they never appear on the intended workspace, but instead on the last initiated / last used 🤦

in i3 config add:
```ini
assign [class="firefox$"] 1: wwww
for_window [class="firefox$"] move container to workspace 1: wwww
```

if that doesnt help, start firefox with `i3_firefox.sh` instead:
```sh
#!/bin/bash
i3-msg 'workspace 1: www'
sleep 1
$(which firefox)
sleep 1
i3-msg 'for_window [class="firefox$"] move container to workspace 1: www'
```

**Comment by RayZ0rr** [comment]
Also experiencing this issue with Firefox. Was going to open an issue
